flink-architecture-tests/src/test/java/org/apache/flink/architecture/rules/ApiAnnotationRules.java

```diff
@@ -153,14 +153,20 @@
                                         @Override
                                         public boolean apply(JavaMethodCall call) {
                                             final JavaClass targetOwner = call.getTargetOwner();
-                                            if (call.getOriginOwner().equals(targetOwner)) {
+                                            final JavaClass originOwner = call.getOriginOwner();
+                                            if (originOwner.equals(targetOwner)) {
                                                 return false;
                                             }
-                                            if (call.getOriginOwner().isInnerClass()
-                                                    && call.getOriginOwner()
-                                                            .getEnclosingClass()
-                                                            .map(targetOwner::equals)
-                                                            .orElse(false)) {
+                                            if (originOwner
+                                                    .getEnclosingClass()
+                                                    .map(targetOwner::equals)
+                                                    .orElse(false)) {
+                                                return false;
+                                            }
+                                            if (targetOwner
```

Comment:
Initially this also checked `isInnerClass()`, but then no violation was removed. Seems like this only returns true if the class is not static, which was a bit surprising.